tools/Az.Tools.Predictor/Az.Tools.Predictor.SourceGenerator/Az.Tools.Predictor.SourceGenerator.csproj

```diff
@@ -0,0 +1,47 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
+    <AssemblyName>Microsoft.Azure.PowerShell.Tools.AzPredictor.SourceGenerator</AssemblyName>
+    <RootNamespace>Microsoft.Azure.PowerShell.Tools.AzPredictor.SourceGenerator</RootNamespace>
+    <RepoArtifacts>$(MSBuildThisFileDirectory)</RepoArtifacts>
+    <OutputPath>$(RepoArtifacts)..\..\..\artifacts\Tools\Az.Tools.Predictor.SourceGenerator\</OutputPath>
+  </PropertyGroup>
+
+  <PropertyGroup>
+      <!--
+         - IsRoslyComponent helps to debug the source generator in Visual Studio. To debug it:
+         - 1. Install the component ".NET compiler platform SDK"
+         - 2. Set the this project as startup project
+         - 3. Configure this project's debug launch profiles: Add an new "Roslyn Component" profile and set the Target Project to Az.Tools.Predictor
+         -->
+    <IsRoslynComponent>true</IsRoslynComponent>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <DocumentationFile>$(OutputPath)\Microsoft.Azure.PowerShell.Tools.AzPredictor.SourceGenerator.xml</DocumentationFile>
+    <PackageId>Az.Tools.Predictor.SourceGenerator</PackageId>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.0.1" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.3" PrivateAssets="all" />
+  </ItemGroup>
+
+  <!-- The generator projects specific dependencies needs to be added to PackageReference and TargetPathWithTargetPlatformMonikor -->
+
+  <ItemGroup>
+    <PackageReference Include="System.Text.Json" Version="6.0.0" GeneratePathProperty="true" PrivateAssets="all"/>
+  </ItemGroup>
+
+  <PropertyGroup>
+    <GetTargetPathDependsOn>$(GetTargetPathDependsOn);GetDependencyTargetPaths</GetTargetPathDependsOn>
+  </PropertyGroup>
+
+  <Target Name="GetDependencyTargetPaths" AfterTargets="ResolvePackageDependenciesForBuild">
+    <ItemGroup>
+      <!-- See potential transient dependency workaround here https://turnerj.com/blog/the-pain-points-of-csharp-source-generators -->
+      <TargetPathWithTargetPlatformMoniker Include="@(ResolvedCompileFileDefinitions)" IncludeRuntimeDependency="false" />
```

Comment:
Maybe a small comment here describing the issue that this workaround is solving as I am sure we will run into it again if we use this generation feature haha